types/angular/index.d.ts

```diff
@@ -1520,23 +1520,27 @@ declare namespace angular {
         (data: T, status: number, headers: IHttpHeadersGetter, config: IRequestConfig): void;
     }
 
-    interface IHttpPromiseCallbackArg<T> {
-        data?: T;
-        status?: number;
-        headers?: IHttpHeadersGetter;
-        config?: IRequestConfig;
-        statusText?: string;
+    interface IHttpResponse<T = any> {
+        data: T;
+        status: number;
+        headers: IHttpHeadersGetter;
+        config: IRequestConfig;
```

Comment:
I digged into the history, and it turned out there was time when this type was used for result values of all `ng.IPromise` promises, not only those returned by `$http`. Since then the properties remained optional.